plugin/src/test/scala/com/github/sbt/jni/javah/NativeMethodTests.scala

```diff
@@ -0,0 +1,44 @@
+package com.github.sbt.jni.javah
+
+import com.github.sbt.jni.javah.util.ClassMetaInfo
+import org.objectweb.asm.ClassReader
+import org.scalatest.funspec.AnyFunSpec
+import scala.jdk.CollectionConverters.asScalaBuffer
+
+import java.io.File
+import java.nio.file.Files
+
+class NativeMethodTests extends AnyFunSpec {
+  this.
+    describe("NativeMethods") {
+      describe("when a method is overloaded") {
+        it("should name methods correctly") {
+          val expectedNames = Set("doSomething__Lcom_github_sbt_jni_samples_A_2",
+            "doSomething__Lcom_github_sbt_jni_samples_B_2",
+            "doSomething___3Lcom_github_sbt_jni_samples_C_2",
+            "doSomething__I_3Lcom_github_sbt_jni_samples_C_2Lcom_github_sbt_jni_samples_B_2D")
+          val methods = asScalaBuffer(NativeMethodTests.classMeta.methods)
+          for (m <- methods) {
+            assert(expectedNames.contains(m.longMangledName()))
+          }
+        }
+      }
+    }
+}
+
+object NativeMethodTests {
+  lazy val classMeta: ClassMetaInfo = {
+    val resource = getClass.getClassLoader.getResource("classes/com/github/sbt/jni/samples/Overloads.class").getPath
```

Comment:
I believe it is using the class loader for NativeMethodsTests.  The getClass call returns the class of NativeMethodsTests.  To use the classes getResource function, which may be what you meant,  The class file would need to be under the same package as the tests.  This can be done.

Reply:
I meant that the anonymous class (block `{ }`) class path sometimes may be different. 

I suggested replacing `getClass.getClassLoadergetResource` to `NativeMethodTests.getClass.getResource`.

Yes, we don't need `Overloads.class` anywhere in this PR, so tests don't pass nor in CI / nor on my machine.

Reply:
Maybe we should move the Overloads.scala file into the test package.  I think this will compile when the test compiles and leave a file where it can be accessed using the NativeMethodTests.getClass.getResource